File: kotlin/src/test/java/it/gabrieletondi/telldontaskkata/doubles/TestShipmentService.kt
package it.gabrieletondi.telldontaskkata.doubles

import it.gabrieletondi.telldontaskkata.domain.Order
import it.gabrieletondi.telldontaskkata.service.ShipmentService

class TestShipmentService : ShipmentService {
    var shippedOrder: Order? = null
        private set

    override fun ship(order: Order) {
        shippedOrder = order
    }
}

File: scala/src/test/scala/doubles/TestShipmentService.scala
package doubles

import ordershipping.domain.Order
import ordershipping.service.ShipmentService

class TestShipmentService extends ShipmentService {
  private var orderToShip: Order = _

  override def ship(order: Order): Unit = orderToShip = order

  def shippedOrder(): Order = orderToShip
}
